### PR TITLE
Lassen (LLNL): ADIOS2, BLAS++ & LAPACK++

### PR DESCRIPTION
**Pull request by ax3l**
Document how to install ADIOS2 (for openPMD `.bp`) and BLAS++/LAPACK++ (for PSATD in RZ) on Lassen at LLNL.

### To Do

- [x] testing after maintenance window ends

### CODE REVIEW

**Comment by EZoni** [comment]
@ax3l @joshualudwig8 Just checking, has this been tested yet?

File: Docs/source/install/hpc/lassen.rst [hunk 1]
```diff
@@ -37,6 +37,34 @@ We recommend to store the above lines in a file, such as ``$HOME/lassen_warpx.pr
...
+   # c-blosc (I/O compression)
+   git clone -b v1.21.1 https://github.com/Blosc/c-blosc.git src/c-blosc
+   rm -rf src/c-blosc-lassen-build
+   cmake -S src/c-blosc -B src/c-blosc-lassen-build -DBUILD_TESTS=OFF -DBUILD_BENCHMARKS=OFF -DDEACTIVATE_AVX2=OFF -DCMAKE_INSTALL_PREFIX=$HOME/sw/lassen/c-blosc-1.21.1
+   cmake --build src/c-blosc-lassen-build --target install --parallel 16
+
+   # ADIOS2
+   git clone -b v2.7.1 https://github.com/ornladios/ADIOS2.git src/adios2
+   rm -rf src/adios2-lassen-build
+   cmake -S src/adios2 -B src/adios2-lassen-build -DADIOS2_USE_Blosc=ON -DADIOS2_USE_Fortran=OFF -DADIOS2_USE_Python=OFF -DADIOS2_USE_ZeroMQ=OFF -DCMAKE_INSTALL_PREFIX=$HOME/sw/lassen/adios2-2.7.1
```
Comment (ax3l) [inline Docs/source/install/hpc/lassen.rst]: No need for SST yet and the found `/usr/lib64/libfabric.so` will most likely not be the one we need (once we need it, we can check which one is fine):
```suggestion
   cmake -S src/adios2 -B src/adios2-lassen-build -DADIOS2_USE_Blosc=ON -DADIOS2_USE_Fortran=OFF -DADIOS2_USE_Python=OFF -DADIOS2_USE_SST=OFF -DADIOS2_USE_ZeroMQ=OFF -DCMAKE_INSTALL_PREFIX=$HOME/sw/lassen/adios2-2.7.1
```

File: Docs/source/install/hpc/lassen.rst [hunk 2]
```diff
@@ -37,6 +37,34 @@ We recommend to store the above lines in a file, such as ``$HOME/lassen_warpx.pr
...
+   # BLAS++ (for PSATD+RZ)
+   git clone https://bitbucket.org/icl/blaspp.git src/blaspp
+   rm -rf src/blaspp-lassen-build
+   cmake -S src/blaspp -B src/blaspp-lassen-build -Duse_openmp=ON -Dgpu_backend=CUDA -Duse_cmake_find_blas=ON -DCMAKE_CXX_STANDARD=17 -DCMAKE_INSTALL_PREFIX=$HOME/sw/lassen/blaspp-master
+   cmake --build src/blaspp-lassen-build --target install --parallel 16
+
+   # LAPACK++ (for PSATD+RZ)
+   git clone https://bitbucket.org/icl/lapackpp.git src/lapackpp
+   rm -rf src/lapackpp-lassen-build
+   cmake -S src/lapackpp -B src/lapackpp-lassen-build -Duse_cmake_find_lapack=ON -DCMAKE_CXX_STANDARD=17 -Dbuild_tests=OFF -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON -DCMAKE_INSTALL_PREFIX=$HOME/sw/lassen/lapackpp-master
```
Comment (ax3l) [inline Docs/source/install/hpc/lassen.rst]:
```suggestion
   cmake -S src/lapackpp -B src/lapackpp-lassen-build -Duse_cmake_find_lapack=ON -DBLA_VENDOR=IBMESSL -DCMAKE_CXX_STANDARD=17 -Dbuild_tests=OFF -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON -DCMAKE_INSTALL_PREFIX=$HOME/sw/lassen/lapackpp-master
```

File: Docs/source/install/hpc/lassen.rst [hunk 3]
```diff
@@ -37,6 +37,34 @@ We recommend to store the above lines in a file, such as ``$HOME/lassen_warpx.pr
...
+   # ADIOS2
+   git clone -b v2.7.1 https://github.com/ornladios/ADIOS2.git src/adios2
+   rm -rf src/adios2-lassen-build
+   cmake -S src/adios2 -B src/adios2-lassen-build -DADIOS2_USE_Blosc=ON -DADIOS2_USE_Fortran=OFF -DADIOS2_USE_Python=OFF -DADIOS2_USE_ZeroMQ=OFF -DCMAKE_INSTALL_PREFIX=$HOME/sw/lassen/adios2-2.7.1
+   cmake --build src/adios2-lassen-build --target install -j 16
+
+   # BLAS++ (for PSATD+RZ)
+   git clone https://bitbucket.org/icl/blaspp.git src/blaspp
+   rm -rf src/blaspp-lassen-build
+   cmake -S src/blaspp -B src/blaspp-lassen-build -Duse_openmp=ON -Dgpu_backend=CUDA -Duse_cmake_find_blas=ON -DCMAKE_CXX_STANDARD=17 -DCMAKE_INSTALL_PREFIX=$HOME/sw/lassen/blaspp-master
```
Comment (ax3l) [inline Docs/source/install/hpc/lassen.rst]:
```suggestion
   cmake -S src/blaspp -B src/blaspp-lassen-build -Duse_openmp=ON -DBLA_VENDOR=IBMESSL -Dgpu_backend=CUDA -Duse_cmake_find_blas=ON -DCMAKE_CXX_STANDARD=17 -DCMAKE_INSTALL_PREFIX=$HOME/sw/lassen/blaspp-master
```

**Comment by ax3l** [comment]
@EZoni I cannot reach @joshualudwig8 right now, but I updated and tested this and all is working condition now :)
Ready to be merged to update our docs :)

**Review by EZoni — APPROVED** [review]
Thank you for the update and for testing, @ax3l! Looks good to me.

**Pull request merged**